string.c

```diff
@@ -4308,37 +4308,34 @@ rb_str_append_as_bytes(int argc, VALUE *argv, VALUE str)
 
 /*
  *  call-seq:
- *    string << object -> string
+ *    self << object -> self
  *
- *  Concatenates +object+ to +self+ and returns +self+:
+ *  Appends a string representation of +object+ to +self+;
+ *  returns +self+.
+ *
+ *  If +object+ is a string, appends it to +self+:
  *
  *    s = 'foo'
  *    s << 'bar' # => "foobar"
  *    s          # => "foobar"
  *
- *  If +object+ is an Integer,
- *  the value is considered a codepoint and converted to a character before concatenation:
+ *  If +object+ is an integer,
+ *  its value is considered a codepoint;
+ *  converts the value to a character before concatenating:
  *
  *    s = 'foo'
  *    s << 33 # => "foo!"
  *
- *  If that codepoint is not representable in the encoding of
- *  _string_, RangeError is raised.
- *
- *    s = 'foo'
- *    s.encoding              # => <Encoding:UTF-8>
- *    s << 0x00110000         # 1114112 out of char range (RangeError)
- *    s = 'foo'.encode(Encoding::EUC_JP)
- *    s << 0x00800080         # invalid codepoint 0x800080 in EUC-JP (RangeError)
```

Comment:
We lost information about this exception case, is that deliberate?

Reply:
Yes.  In general, the [doc guide](https://docs.ruby-lang.org/en/master/contributing/documentation_guide_md.html#label-Corner+Cases+and+Exceptions) discourages mentioning exceptions.  In this case, I thought the error message for the raised exception explains itself.

Reply:
The doc guide states:

> For corner cases of methods, such as atypical usage, briefly mention the behavior, but do not provide any examples.
>
> Only document exceptions raised if they are not obvious.

The error message might be obvious, but I don't think it's obvious what the behavior is for this exceptional case.

Reply:
Restored.